src/model/mod.rs

```diff
@@ -235,12 +236,94 @@ fn valid_field(player: &types::Player, input: usize, ori: &str) -> bool {
             }
         },
     }
-
     valid
 }
 
+/// Removes the current rand value from the remaining possibilities.
+/// Since every call of remove() shifts every remaining element to the left,
+/// we need to search for the index of the element to be deleted.
+fn remove_idx(rand: usize, vec: &mut Vec<usize>) {
+    for i in 0..vec.len() - 1 {
+        if vec[i] == rand {
+            vec.remove(i);
+        }
+    }
+}
+
+/// Random ship placement for the AI.
+fn place_ai(player: &mut types::Player, ship: &types::ShipType,
+    mut vec: &mut Vec<usize>) -> Result<(), types::ErrorType> {
```

Comment:
It looks like it has been [decided 14 hours ago](https://github.com/rust-lang-nursery/fmt-rfcs/issues/39#issuecomment-278544439). I think this means the right formatting for you is:

```
fn place_ai(
    player: &mut types::Player, 
    ship: &types::ShipType,
    mut vec: &mut Vec<usize>,
) -> Result<(), types::ErrorType> {
    ...
}
```